SQL/proc_descriptions/procedure_compute_speeds_for_segments.md

```diff
@@ -3,29 +3,61 @@
+        - Insert the merged data into `nodes_ways_speeds`.
+
+This procedure is designed to efficiently calculate and update speed records for given time parameters in a specified target area. The use of temporary tables and indexes ensures optimal performance during the data processing steps.
 
 ## TODO list
 - [x] Create description of the procedure
-- [x] Check if index `target_ways_id_idx` of `target_ways` is used. Is used
-- [x] Check if indexes `node_segments_osm_id_idx`, `node_segments_wf_idx`, `node_segments_wt_idx` of `node_sequences` table are used. Only `node_segments_wf_idx` is used -> Issue.
+- [x] Check if index `target_ways_id_idx` of `target_ways` is used. __Is indeed used__
+- [x] Check if indexes `node_segments_osm_id_idx`, `node_segments_wf_idx`, `node_segments_wt_idx` of `node_sequences` table are used. __Only `node_segments_wf_idx` is used__ -> Issue.
 - [x] Add deeper exaplanation of data selection for `grouped_speed_records` TEMP table. - No need, selection is plain.
-- [x] Replace `..., ...` in explanation of flow with classification name.
-- [ ] Reduction of TEMP tables into CTEs AKA With statements. - `target_ways`, `node_sequences` are candidates, not sure about `grouped_speed_records`.
-- [ ] Reduction of If-else statement with conditional WHERE clause with conditional WHERE clause. - Not easily achived as we have two different source for every case of dataset_quality, and refactored block with great chance would have worse perfomance. Does not worth it, in my opinion.
-- [ ] Reduction of calculation of average speed in 3rd step of main flow - it is calculated twice. May influence the execution perfomance depending on the number of stored data. - Not really sure, cause window function avg() applies after finding records, not before.
+- [ ] Reduction of TEMP tables into CTEs AKA With statements. - `target_ways` - used twice in independent contexts (first one for __STDOUT output__) + additional INDEXes, `node_sequences` - used twice in independent contexts (first one for __STDOUT output__) + 3 INDEXes, `grouped_speed_records` - same as previous two + different table sources are applied to different input of function.
+- [ ] Reduction of If-else statement (1. step) with conditional WHERE clause. - Not easily achived as we have two different source for every case of dataset_quality, and refactored block with great chance would have worse perfomance. Does not worth it, in my opinion.
+<!-- - [ ] Reduction of calculation of average speed in 3rd step of main flow - it is calculated twice. May influence the execution perfomance depending on the number of stored data. - Not really sure, cause window function avg() applies after finding records, not before. -->
+<!-- On the other hand, im not really sure why i've brought it up -->
+- [x] Humanize/formalize the description.
 
 ## QA
 - Indexes `node_segments_osm_id_idx` & `node_segments_wt_idx` are not used, I think we should remove their creation queries.
```

Comment:
If these columns are not present in any of the queries, I agree.

Reply:
These 2 particular indices are a combination of 2 columns:
```
CREATE INDEX node_segments_wf_idx ON node_sequences(way_id, from_position);
CREATE INDEX node_segments_wt_idx ON node_sequences(way_id, to_position);
```
Either way, I've checked with `Explain` query if it is used and if I remember correctly, the result showed no usage of these indices

Reply:
If there is any JOIN on these pairs of columns, I suggest to keep the indices, regardless of EXPLAIN

Reply:
```sql
			 JOIN node_sequences ns
				  ON speed_records.from_osm_id = ns.from_id
					  AND speed_records.to_osm_id = ns.to_id
					  AND from_position > to_position
```
Does this Join means that we use pair (from_id, to_id)?

Reply:
Okay, I've referted commit, in which they suspicious indices were removed